### PR TITLE
Association

### PR DESCRIPTION
**Pull request by kennyegun24**
In this PR, I did the following
Added Classroom, Book and Rental class
Added a has_many and belong_to relationship between Classroom and Student class
Added relationships between Person and Book using the class Rental

### CODE REVIEW

**Comment by omarsalem7** [comment]
Hi @kennyegun24 😊, 

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 💀 

You have submitted a project, that has falling linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+  def initialize(classroom)
+    super('kenny', 10)
     @classroom = classroom
   end
 
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def classrooms=(classroom)
```
Comment (sja-thedude) [inline student.rb]: - [ ] You have made a typo here its classroom, not classrooms.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :author, :title, rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
```
Comment (sja-thedude) [inline book.rb:10]: - [ ] Kindly add a method for rentals as it's required

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @kennyegun24, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters are passing
✔️ Correct Gitflow
✔️ Professional Readme

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (kennyegun24) [reply student.rb]: Good evening @sja-thedude, it gives me an error since i had already defined classroom already on line 7

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @kennyegun24      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R





 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

File: student.rb (hunk 1 above)
Comment (sja-thedude) [reply student.rb]: Thats fine, I thought its typo

**Pull request merged**